### ISSUE TITLE
Data-Only container with 1.2 rc2 broken on first "up"

### ISSUE THREAD

**Issue by StevenWeathers** [docker/compose#1193]
I've setup a project that has a "data-only" container e.g. a container that defines a volume that is then "volumes_from" a parent application.  This was working just fine in 1.1.0, however when I switched to 1.2 rc2 it fails on first "up", but works on second "up" (recreate vs create).

**Comment by aanand** [comment]
Thanks for reporting. Could you post a minimal failing `docker-compose.yml` and the error you get?

**Comment by StevenWeathers** [comment]
Sure, here's the error and I'll get you a working example asap.

Creating data_1...
Creating web_1...
Cannot start container 7bb781756791616de10fc61f5f8c3928ffe91ccdb7533a3b5037763925a993a3: Cannot link to a non running container: /data_1 AS /web_1/webdata

**Comment by StevenWeathers** [comment]
Just an fyi, I mentioned in another issue that I had RC1 installed, just tried RC2 with this issue and it still exists.

```
web:
  build: ./web
  links:
    - data
  ports:
    - "80:80"
  volumes_from:
    - data

data:
  build: ./data
```

**Comment by StevenWeathers** [comment]
Here's the Dockerfile from data

```
FROM ubuntu:trusty
RUN mkdir /usr/local/web
VOLUME ["/usr/local/web"]
CMD /bin/sh
```

**Comment by aanand** [comment]
Just so we can narrow it down, does the error still happen if you replace your `build` keys with `image: busybox` and `command: sleep 300`?

**Comment by StevenWeathers** [comment]
I will try that asap and get back to you.

**Comment by rcoedo** [comment]
Same problem here with a similar setup. Replacing `build` with `image` and using `sleep 300` as command didn't fix the problem.

**Comment by killerwolf** [comment]
the problem is that data container stops and compose "exits", it won't create the following containers (the ones linked to data containers)

To circumvents this issue, i added `command: tail -f /dev/null` , this way the data container keeps running


```
mydata:
  image: killerwolf/data
  command: tail -f /dev/null
myscrit:
  build: .
  volumes_from:
   - ndsdata
```

**Comment by GuillaumeSeren** [comment]
Hi there,
adding 'command: tail -f /dev/null' to the related container fix the error.

In my case it was a DB container.

**Comment by retgoat** [comment]
Same for me.
```
db:
  image: mysql
  command: tail -f /dev/null
```
Fix the error

**Comment by yoanisgil** [comment]
tail -f /dev/null does not work for me, see my services definition here: 

````
shell:
  image: domainsetup-service:latest
  working_dir: /srv/www
  entrypoint: ["/bin/bash"]
  stdin_open: true
  tty: true
  volumes:
    - .:/srv/www
    - ~/.ssh:/root/.ssh
    - ./docker/php.d/15-xdebug.ini:/etc/php.d/15-xdebug.ini
    - ~/.composer:/root/.composer
  env_file:
    - ./docker/config.env
  links:
    - powerdns:powerdns.local
powerdns:
    image: crakmedia/powerdns-authoritative:3.4.5
    volumes:
        - ./docker/pdns.d/pdns.gmysql.conf:/etc/pdns/pdns.d/pdns.gmysql.conf
    ports:
        - "553/udp:53/udp"
        - "8081:8081"
    links:
        - pdns_mysql:mysql
    hostname: powerdns
    command: pdns_server
pdns_mysql:
    image: mysql:5.6
    volumes_from:
        - pdns_mysql_data
    ports:
        - "4406:3306"
pdns_mysql_data:
    image: yoanisgil/powerdns-authoritative-schema:345.56
    command: tail -f /dev/null
```

and when I run ```docker-compose run --rm shell -c "bin/phpunit tests/Functional"```

```
Creating domainsetupservice_pdns_mysql_data_1...
Creating domainsetupservice_pdns_mysql_1...
Creating domainsetupservice_powerdns_1...
Cannot start container af8dd69aec37bbb6a2ab4678d5e6f17eb72e176dec0311cb95ee35509e56cafd: Cannot link to a non running container: /domainsetupservice_pdns_mysql_1 AS /domainsetupservice_powerdns_1/domainsetupservice_pdns_mysql_1
```

**Comment by aanand** [comment]
It looks like your `pdns_mysql ` container is exiting before `powerdns` can link to it. Have a look at its logs maybe:

    $ docker-compose logs pdns_mysql

**Comment by yoanisgil** [comment]
Yes. I also had to add an explicit link to pdns_mysql_data.  After that the tail hack did the trick ;)

**Comment by dnephin** [comment]
This should be addressed in 1.4.x since we now wait for all containers to exit, not just the first one.
